lib/puppet/network/http/factory.rb

```diff
@@ -23,10 +23,13 @@ def create_connection(site)
     Puppet.debug("Creating new connection for #{site}")
 
     args = [site.host, site.port]
-    if Puppet[:http_proxy_host] == "none"
-      args << nil << nil
-    else
-      args << Puppet[:http_proxy_host] << Puppet[:http_proxy_port]
+
+    unless Puppet::Util::HttpProxy.no_proxy?(site)
+      if Puppet[:http_proxy_host] == "none"
+        args << nil << nil
+      else
+        args << Puppet[:http_proxy_host] << Puppet[:http_proxy_port]
```

Comment:
It appears puppet's http connection factory doesn't handle `http_proxy_user` and `http_proxy_password`. Could you file a new ticket for that? And even submit a separate PR for that?